pkg/network/controllers/vm_test.go

```diff
@@ -244,6 +244,56 @@ var _ = Describe("VM Network Controller", func() {
 		Expect(iface.State).NotTo(Equal(v1.InterfaceStateAbsent))
 	})
 
+	DescribeTable("sync updates link state correctly", func(fromState, toState v1.InterfaceState, expectChange bool) {
+		clientset := fake.NewSimpleClientset()
+		c := controllers.NewVMController(
+			clientset,
+			stubPodGetter{pod: &k8sv1.Pod{}},
+		)
+
+		vmi := libvmi.New(
+			libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+			libvmi.WithNetwork(v1.DefaultPodNetwork()),
+		)
+
+		vmi.Status.Interfaces = append(vmi.Status.Interfaces, v1.VirtualMachineInstanceNetworkInterface{Name: "default"})
```

Comment:
nit: libvmi could do it as well, please see:
https://github.com/kubevirt/kubevirt/blob/8ac93a9605f499cd7a878321e174ae75122cd44e/pkg/network/setup/filters_test.go#L71